### ISSUE TITLE
How can I supply a whole custom instance as a field?

### ISSUE THREAD

**Issue by scardonameli** [instancio/instancio#171]
Hello everyone!

First I want to thank you for this awesome library, it's so useful and semantic. I have a question, I was trying to build an instance from my record `PC`, but I couldn't create it using _strict mode_. I wrote something like this:

```java
import static org.instancio.Select.field;

import java.util.List;
import org.instancio.Instancio;
import org.instancio.Model;
import org.instancio.Random;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class IssueTest {
  public enum Socket { AMD, INTEL}
  public record CPU(String model, Double frequency){}
  public record RAM(String model, Integer capacity){}
  public record MainBoard(String model, Socket socket, String chipset){}

  record PC(MainBoard board, CPU cpu, List<RAM> memories){}

  static CPU customCPU(Random random){
    String model = random.alphanumeric(10).concat("_cpu");
    Double frequency = random.doubleRange(3.0, 5.0);
    return new CPU(model, frequency);
  }

  static MainBoard customMainBoard(Random random){
    String model = random.alphanumeric(10).concat("_board");
    Socket socket = random.oneOf(Socket.values());
    String chipset = random.alphanumeric(5).concat("_socket");

    return new MainBoard(model, socket, chipset);
  }

  static List<RAM> customMemories(Random random){
    String model = random.alphanumeric(15).concat("_ram");
    Integer capacity = random.oneOf(4, 8, 16, 32);
    RAM memory = new RAM(model, capacity);
    return List.of(memory);
  }

  @Test
  void customisedPC() {
    Model<PC> model =  Instancio.of(PC.class)
        .supply(field("cpu"), IssueTest::customCPU)
        .supply(field("board"), IssueTest::customMainBoard)
        .supply(field("memories"), IssueTest::customMemories)
        .lenient() //It's necessary to run.
        .toModel();

    PC pc = Instancio.create(model);
    System.out.println(pc);
    
    Assertions.assertTrue(pc.memories.get(0).model.contains("_ram"), "It wasn't generated by a custom generator.");
    Assertions.assertFalse(pc.cpu.model.contains("_cpu"), "It was generated by a custom generator.");
  }

}

```

Is possible to supply a whole custom field? I saw that if I use generators for one-by-one of the fields on the `PC` then it would work.  Thanks in advance.

**Comment by armandino** [comment]
Thank you for the feedback and the question! It is much appreciated.

It seems that you've discovered a bug related to how records are handled. I did a quick test changing the `CPU` from `record` to `class` and it worked as expected.

I will investigate this more and create a bug fix to address this issue. I'll let you know once it's fixed (hopefully next few days).

**Comment by scardonameli** [comment]
Thanks, @armandino for your quick answer. I'll be pending for your feedback 👌🏼

**Comment by armandino** [comment]
Hi @scardonameli, this bug should be fixed in version `1.5.2`. I just released it to Maven central, so it should be available in a few hours. Thanks again for reporting this issue!
